src/multiselect_controls.js

```diff
@@ -412,6 +412,13 @@ export class MultiselectControls {
         this.updateDraggables_(this.lastSelectedElement_);
         this.lastSelectedElement_ = null;
         inPasteShortcut.set(this.workspace_, false);
+      } else if (!this.dragSelection.size && !(Blockly.getSelected() instanceof
+          MultiselectDraggable)) {
+        if (Blockly.getSelected() instanceof Blockly.BlockSvg &&
+            !Blockly.getSelected().isShadow()) {
+          Blockly.common.setSelected(null);
+        }
+        this.justUnselectedBlock_ = Blockly.getSelected();
```

Comment:
Is it right to set `this.justUnselectedBlock_` even if we are not unselecting any blocks, if the branch above doesn't get hit?

Reply:
Yes, this is because right after this branch is called, the block gets unselected. The setSelected call is called twice even if we only click once. I think this is because of the gesture handling thing that Beka was talking about.

Reply:
Okay, then I would say this is what might be broken in the future once the gesture update is complete, it would be good to add a to do comment here documenting the situation.